codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddTranscribeStreamingDependency.java

```diff
@@ -92,6 +94,11 @@ private static boolean isTranscribeStreaming(ServiceShape service) {
         String serviceId = service.getTrait(ServiceTrait.class).map(ServiceTrait::getSdkId).orElse("");
         return serviceId.equals("Transcribe Streaming");
     }
+
+    private static boolean hasEventStreamInput(Model model, ServiceShape service, OperationShape operation) {
+        EventStreamIndex eventStreamIndex = EventStreamIndex.of(model);
+        return eventStreamIndex.getInputInfo(operation).isPresent();
+    }
```

Comment:
this is ok as a patch, but long term the conditional method should be exported for use by the websocket plugin, since it requires this plugin to be present.

Reply:
PR posted at https://github.com/aws/aws-sdk-js-v3/pull/6351